File: dcp-solutions/2024-12-13/solution.js
```javascript
class TreeNode {
    constructor(val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

function maxPathSum(root) {
    let res = -Infinity;

    function dfs(node, current_sum) {
        if (!node) return 0;
        current_sum += node.val;
        let left = Math.max(dfs(node.left, current_sum), 0);
        let right = Math.max(dfs(node.right, current_sum), 0);
        res = Math.max(res, current_sum + left + right);
        return Math.max(left, right);
    }

    dfs(root, 0);
    return res;
}
```

File: dcp-solutions/2024-12-13/solution.rb
```ruby
# Definition for a binary tree node.
class TreeNode
  attr_accessor :val, :left, :right
end

def max_path_sum(root)
  def dfs(node)
    return -Infinity if not node
    left_max = -Infinity
    right_max = -Infinity
    if node.left
      left_max = [node.val + dfs(node.left) , left_max].max
    end
    if node.right
      right_max = [node.val + dfs(node.right), right_max].max
    end
    return [node.val + (left_max > 0 ? left_max : 0) + (right_max > 0 ? right_max : 0), -Infinity].max
  end

  return dfs(root)
end
```